src/lib.rs

```diff
@@ -60,6 +71,62 @@ pub fn parse(contents: &str) -> Result<EditorConfigINIAST, Error<Rule>> {
 							}
 						}
 					}
+					Rule::comment => {
+						let mut inner_rules = line.into_inner();
+						let comment = Item::Comment(Comment {
+							indicator: inner_rules.next().unwrap().as_str().to_string(),
+							value: inner_rules.next().unwrap().as_str().to_string(),
+							raws: Raws {
+								before: Some("".to_string()),
+								after: Some("".to_string()),
+								newline: Some(inner_rules.next().unwrap().as_str().to_string()),
+							},
+						});
+						match &mut current_section {
+							Some(ref mut section) => {
+								section.body.push(comment);
+							}
+							None => {
+								ast.body.push(comment);
+							}
+						}
+					}
+					Rule::blank_line => {
+						let mut inner_rules = line.into_inner();
+						let whitespace = inner_rules.next().unwrap().as_str().to_string();
+						let line = Item::Raws(Raws {
+							before: if whitespace == "" {
+								Some(whitespace)
+							} else {
+								None
+							},
+							after: None,
+							newline: Some(inner_rules.next().unwrap().as_str().to_string()),
+						});
+						match &mut current_section {
+							Some(ref mut section) => {
+								section.body.push(line);
+							}
+							None => {
+								ast.body.push(line);
+							}
+						}
+					}
+					Rule::ws => {
```

Comment:
This rule is here solely for the single scenario in which the file has whitespace w/o a final newline at the end. Couldn't figure out how to do it without.

Reply:
If i get it right you're not happy with it. I think this is too complicated, too - we will find a better way to solve that in a while.